File: packages/react-native-fast-io/ios/HybridCompressorFactory.swift
//
//  HybridCompressorFactory.swift
//  FastIO
//
//  Created by Mike Grabowski on 11/11/2024.
//

import Foundation

class HybridCompressorFactory : HybridCompressorFactorySpec {
  func create(algorithm: CompressionAlgorithm) throws -> (any HybridCompressorSpec) {
    return HybridCompressor(algorithm: algorithm)
  }
  
  var hybridContext = margelo.nitro.HybridContext()
  var memorySize: Int {
    return getSizeOf(self)
  }
}

File: packages/react-native-fast-io/android/src/main/java/com/margelo/nitro/fastio/HybridCompressorFactory.kt
package com.margelo.nitro.fastio

class HybridCompressorFactory : HybridCompressorFactorySpec() {
    override fun create(algorithm: CompressionAlgorithm): HybridCompressorSpec {
        throw NotImplementedError("HybridCompressorFactory.create() not implemented")
    }

    override val memorySize: Long
        get() = 0L
} 
